### PR TITLE
Feature/chat localisation

### PR DESCRIPTION
**Pull request by DeepGreen1**

### CODE REVIEW

File: app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt [hunk 1]
```diff
@@ -56,4 +62,27 @@ class ChatActivityTest {
...
+        )
+    }
+
+    // TODO: See with the team if granting permissions only during tests is a good idea (better coverage)
+    //@get:Rule var permissionRule1: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_COARSE_LOCATION)
+    //@get:Rule var permissionRule2: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+    //@Test
+    //fun localisationPermissionAsked() {
+    //    onView(withId(R.id.buttonSendLocalisation)).perform(click())
+    //}
```
Comment (omarelmalki) [inline app/src/androidTest/java/com/github/brugapp/brug/ChatActivityTest.kt:87]: As long as the code on the repository is the code of a working application (no features are missing), we can try to do that in my opinion.
Don't forget to delete or uncomment this code once we have decided.

**Review by omarelmalki — COMMENTED** [review]
Good job on the location. Feel free to checkout the few comments I left.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 2]
```diff
@@ -39,8 +51,57 @@ class ChatActivity : AppCompatActivity() {
...
+                0.1f,
+                locationListener
+            )
+            // Stop the update as we only want it once (at least for now)
+            locationManager.removeUpdates(locationListener)
+        }
+    }
+
+    // Ask ViewModel to compute UI updates
+    @RequiresApi(Build.VERSION_CODES.O) // Needed for the datetime
```
[the remark this answers is not in the document]
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:94]: Yes! Oreo has the API number 27 and we are devopping for API 31 if I'm not wrong. However, Android Studio asked me to add this annotation, so I added it. I think that changes nothing for later versions.

[the remark this answers is not in the document]
Comment (omarelmalki) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:94]: I see, thank you for your answer.

[the remark this answers is not in the document]
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:94]: Yeah, let's do this! Thanks

**Comment by DeepGreen1** [comment]
> Great job, but make sure to have at least 85% coverage in your tests. After that, it will be an approval for me, but we will have to adapt your code a little bit to only use local classes instead of Firebase (my part doesn't use it due to the previous work I did on the ChatMenuActivity, but no worries your code will still be there :) )

Well, on my PC I had 76% coverage. The remaining part seems to concern the autorisation parts (I'll let you check the comment that I wrote in the ChatActivityTest). Strangely, it's not the same percentage on Cirrus; I'll have to make more research about this.
We can discuss the autorisation issue after the meeting and I can create a task for the next sprint in order to improve the test coverage.

**Comment by Mouniraki** [comment]
> > Great job, but make sure to have at least 85% coverage in your tests. After that, it will be an approval for me, but we will have to adapt your code a little bit to only use local classes instead of Firebase (my part doesn't use it due to the previous work I did on the ChatMenuActivity, but no worries your code will still be there :) )
> 
> Well, on my PC I had 76% coverage. The remaining part seems to concern the autorisation parts (I'll let you check the comment that I wrote in the ChatActivityTest). Strangely, it's not the same percentage on Cirrus; I'll have to make more research about this. We can discuss the autorisation issue after the meeting and I can create a task for the next sprint in order to improve the test coverage.

Ok fair enough, in this state your code doesn’t comply with my changes, but I will make sure that everything is bound correctly before doing the merge. I approve your PR but don’t forget to add an issue to fix your tests and have a better coverage for next sprint.

**Review by Mouniraki — APPROVED** [review]
Nothing more to add, read my last reply.

**Review by omarelmalki — APPROVED** [review]
Thank you for the precisions. I approve the changes for now but we need to get more coverage during next sprint, it would be great to have a task to do this. Since you have dependencies that limit the testing, I suggest taking a look into Hilt and Mockito to get more testing coverage.

**Pull request merged**